src/agent/rustjail/src/cgroups/fs/mod.rs

```diff
@@ -27,1120 +43,848 @@ macro_rules! sl {
+                }
+                let dr = BlkIoDeviceResource {
+                    major: d.blk.major as u64,
+                    minor: d.blk.minor as u64,
+                    weight: w,
+                    leaf_weight: lw,
+                };
+                vec.push(dr);
+            }
+            res.blkio.weight_device = vec;
+
+            let mut vec = vec![];
+            for d in blkio.throttle_read_bps_device.iter() {
+                let tr = BlkIoDeviceThrottleResource {
+                    major: d.blk.major as u64,
+                    minor: d.blk.minor as u64,
+                    rate: d.rate,
+                };
+                vec.push(tr);
+            }
+            res.blkio.throttle_read_bps_device = vec;
+
+            let mut vec = vec![];
+            for d in blkio.throttle_write_bps_device.iter() {
+                let tr = BlkIoDeviceThrottleResource {
+                    major: d.blk.major as u64,
+                    minor: d.blk.minor as u64,
+                    rate: d.rate,
+                };
+                vec.push(tr);
+            }
+            res.blkio.throttle_write_bps_device = vec;
+
+            let mut vec = vec![];
+            for d in blkio.throttle_read_iops_device.iter() {
+                let tr = BlkIoDeviceThrottleResource {
+                    major: d.blk.major as u64,
+                    minor: d.blk.minor as u64,
+                    rate: d.rate,
+                };
+                vec.push(tr);
+            }
+            res.blkio.throttle_read_iops_device = vec;
+
+            let mut vec = vec![];
+            for d in blkio.throttle_write_iops_device.iter() {
+                let tr = BlkIoDeviceThrottleResource {
+                    major: d.blk.major as u64,
+                    minor: d.blk.minor as u64,
+                    rate: d.rate,
+                };
+                vec.push(tr);
+            }
+            res.blkio.throttle_write_iops_device = vec;
+        }
 
-        if r.cpu.is_some() {
-            let cpu = r.cpu.as_ref().unwrap();
-            cpus = cpu.cpus.as_str();
-            mems = cpu.mems.as_str();
+        if r.hugepage_limits.len() > 0 {
+            info!(sl!(), "cgroup manager set hugepage !!!");
+            res.hugepages.update_values = true;
+            let mut limits = vec![];
+
+            for l in r.hugepage_limits.iter() {
+                let hr = HugePageResource {
+                    size: l.page_size.clone(),
+                    limit: l.limit,
+                };
+                limits.push(hr);
+            }
+            res.hugepages.limits = limits;
         }
 
-        // For updatecontainer, just set the new value
-        if update {
-            if !cpus.is_empty() {
-                try_write_file(dir, CPUSET_CPUS, cpus)?;
+        if r.network.is_some() {
+            info!(sl!(), "cgroup manager set network !!!");
+            let network = r.network.as_ref().unwrap();
+
+            // cls classid
```

Comment:
This comment is incongruous, given there is no other comment of the same type.

If you want to put useful comments, maybe put URLs to some source documentation for each resource? Helping the reader understand what is the range and expected behavior for each setting?